src/ExtDate.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EDTF;
+
+
+class ExtDate
+{
+    protected ?int $year;
+    protected ?int $month;
+    protected ?int $day;
+
+    private Parser $parser;
+
+    public function __construct(Parser $parser)
```

Comment:
Like this you can't construct a `ExtDate` without having a `Parser`. That is really strange, especially after we talked about Value Objects. Why not this?

```php
__construct(?int $year, ?int $month, ?int $day)
```

Or preferably 


```php
__construct(int $year, int $month, int $day)
```

Reply:
Optional fields are very suspect in objects like this. Especially all fields being optional.